apps/ui-library/registry/default/blocks/dropzone/hooks/use-supabase-upload.ts

```diff
@@ -126,9 +126,12 @@ const useSupabaseUpload = (options: UseSupabaseUploadOptions) => {
 
     const responses = await Promise.all(
       filesToUpload.map(async (file) => {
+        // Set a safe filename to avoid issues with special characters
+        const safeFilename = generateSafeFilename(file.name)
+
         const { error } = await supabase.storage
           .from(bucketName)
-          .upload(!!path ? `${path}/${file.name}` : file.name, file, {
+          .upload(!!path ? `${path}/${safeFilename}` : safeFilename, file, {
```

Comment:
would it make more sense if file name sanitization happened in supabase-js instead?
@fenos

Reply:
@ivasilov what you think?